autode/config.py

```diff
@@ -16,360 +16,401 @@
 
 
 class _ConfigClass:
```

Comment:
I tried setting class attributes (i.e. no `self.X` and `_init_finished` flag) and it seems to work okay..(?)

Reply:
I see the problem 😞 – just not a big fan of the solution! will have a play around

Reply:
@t-young31 Something like this  would read very similar to the original config.py:
```python
class _OptionSetterConfig:
  def __init__(self, **kwargs):
    for name, attr in kwargs.items():
      setattr(self, name, attr)

Config = _OptionSetterConfig(
  n_cores = 4,
  max_core = 4000,
  ....
  ORCA = _OptionSetterConfig(
    keywords = KeywordsSet( ... ),
    solvation = solv.smd,
    ...
  G09 = _OptionSetterConfig(
    keywords = ...
  )
  ...
)
```
Would this work? I have tried a lot of other hacks to get pickling to work, but I can't find other ways.

Reply:
@t-young31 Another possible hacky solution to the readability problem would be to use the word Config instead of self to name the class, like this:

```python
class _ConfigClass:
  def __init__(Config):
    Config.n_cores = 4
    ...
  class ORCA:
    def __init__(ORCA):
      ORCA.keywords = ...
```

Reply:
Both I think would be okay. Would the former require the __setattr__ override in the _OptionSetterConfig class, which might clutter things a bit. Latter I think is totally fine. I've tried a potential different option: https://github.com/t-young31/autodE/tree/potential_config_pickle_solution – what do you think ?  (pretty ugly I know but at least the top of the file is clean)

Reply:
@t-young31 Thanks! I looked at the potential solution that you wrote, and I found two ways of making it a bit simpler. It still looks a bit cluttered, but the original ConfigClass does not need to be changed in any way. Could you please check config.py and let me know what you think?

Reply:
thanks! I really like the "alternative solution" that just instantiates (maybe worth renaming it to something more expressive and moving it to the bottom of the file?) 😄 
didn't know you could do `isinstance(<thing>, type):` – awesome

Reply:
@t-young31 Thanks, I have moved the function to the bottom and renamed it. Let me know if the name makes sense, I am generally bad at writing function/class names 😅 . I was previously using `inspect.isclass()` to check for class, but I looked at the source code, and they were using `isinstance(something ,type)` so I used that instead of importing inspect.